tests/topotests/route-scale/test_route_scale.py

```diff
@@ -126,85 +159,36 @@ def test_route_install():
     removed_file = "{}/r1/no.routes.json".format(CWD)
     expected_removed = json.loads(open(removed_file).read())
 
-    r1 = tgen.gears["r1"]
+    # dict keys of params: ecmp number, corresponding nhg name, timeout,
+    # number of times to wait
+    scale_keys = ['ecmp', 'nhg', 'wait', 'count']
 
-    r1.vtysh_cmd("sharp install route 1.0.0.0 nexthop-group one 1000000", isjson=False)
-    test_func = partial(topotest.router_json_cmp, r1, "show ip route summary json", expected_installed)
-    _, result = topotest.run_and_expect(test_func, None, count=40, wait=5)
-    output = r1.vtysh_cmd("sharp data route", isjson=False)
-    logger.info("1 million routes X 1 ecmp installed")
-    logger.info(output)
-    r1.vtysh_cmd("sharp remove route 1.0.0.0 1000000", isjson=False)
-    test_func = partial(topotest.router_json_cmp, r1, "show ip route summary json", expected_removed)
-    _, result = topotest.run_and_expect(test_func, None, count=40, wait=5)
-    output = r1.vtysh_cmd("sharp data route", isjson=False)
-    logger.info("1 million routes x 1 ecmp removed")
-    logger.info(output)
+    # Table of defaults (mainly useful for timeout values)
+    scale_defaults = dict(zip(scale_keys, [None, None, 5, 40]))
 
-    r1.vtysh_cmd("sharp install route 1.0.0.0 nexthop-group two 1000000", isjson=False)
-    test_func = partial(topotest.router_json_cmp, r1, "show ip route summary json", expected_installed)
-    _, result = topotest.run_and_expect(test_func, None, count=40, wait=5)
-    output = r1.vtysh_cmd("sharp data route", isjson=False)
-    logger.info("1 million routes X 2 ecmp installed")
-    logger.info(output)
-    r1.vtysh_cmd("sharp remove route 1.0.0.0 1000000", isjson=False)
-    test_func = partial(topotest.router_json_cmp, r1, "show ip route summary json", expected_removed)
-    _, result = topotest.run_and_expect(test_func, None, count=40, wait=5)
-    output = r1.vtysh_cmd("sharp data route", isjson=False)
-    logger.info("1 million routes x 2 ecmp removed")
-    logger.info(output)
+    # List of params for each step in the test; note extra time given
+    # for the highest ecmp steps
+    scale_steps = [
+        [1, 'one'], [2, 'two'], [4, 'four'],
+        [8, 'eight'], [16, 'sixteen', 5, 60]
```

Comment:
I think we could bump the sixteen case a bit more, at least for address sanitizer which is exceeding the 200 seconds mark.

How about `16, 'sixteen', 7, 57`? It would total to 399 seconds but with an interval 2 seconds longer making FRR spend less time generating JSON output.

Reply:
yes, sure - now that I've had a look at the timings for the x16 run, I'll re-add the x32, and bump the interval for those larger two or three steps.